packages/flutter/lib/src/material/tabs.dart

```diff
@@ -999,6 +999,10 @@ class _TabBarState extends State<TabBar> {
       _controller!.animation!.addListener(_handleTabControllerAnimationTick);
       _controller!.addListener(_handleTabControllerTick);
       _currentIndex = _controller!.index;
+
+      if (widget.isScrollable && _scrollController != null) {
+        _scrollToCurrentIndex();
```

Comment:
Ok I have found why the bounce @goderbauer noticed is happening.

https://user-images.githubusercontent.com/16964204/185234192-d4657dee-8cd3-46d7-b642-e70c048e6300.mov

It is much more noticeable if you make one of the lists of tabs much longer than the other one, shown above.

You may be missing a setState call before calling _scrollToIndex, that may be a good place to look first here. What it looks like is that:

in LongTab, the last index is say 100 pixels away.
in the shorter tab, the last index is closer, maybe 50 pixels away. 

When the index changes here and you call _scrollToIndex, it is using the information from the long tab, rather than the shorter tab. This is why there is a large bounce in the video, it is trying to scroll much further than it needs to.